scheduler/src/cook/kubernetes/compute_cluster.clj

```diff
@@ -231,18 +284,79 @@
+          ; k8s scheduler scheduling those synthetic pods, and we will often lose, resulting in
+          ; Failed instances for no good reason.
+          (log/info "In" name "compute cluster, skipping offer generation for pool" pool-name
+                    "because there are" num-waiting-synthetic-pods
+                    "waiting synthetic pod(s) in that pool")
+          [])
+        (let [offers-all-pools (generate-offers this nodes pods)
+              ; TODO: We are generating offers for every pool here, and filtering out only offers for this one pool.
+              ; TODO: We should be smarter here and generate once, then reuse for each pool, instead of generating for each pool each time and only keeping one
+              offers-this-pool (get offers-all-pools pool-name)]
+          (log/info "In" name "compute cluster, generated" (count offers-this-pool) "offers for pool" pool-name
+                    (into [] (map #(into {} (select-keys % [:hostname :resources])) offers-this-pool)))
+          offers-this-pool))))
 
   (restore-offers [this pool-name offers])
 
+  (autoscaling? [_ pool-name]
+    (-> synthetic-pods-config :pools (contains? pool-name)))
+
+  (autoscale! [this pool-name task-requests]
+    (try
+      (assert (cc/autoscaling? this pool-name)
+              (str "In " name " compute cluster, request to autoscale despite invalid / missing config"))
+      (let [outstanding-synthetic-pods (->> @all-pods-atom
+                                            (add-starting-pods this)
+                                            (filter controller/synthetic-pod->job-uuid))
+            num-synthetic-pods (count outstanding-synthetic-pods)
+            {:keys [image user command max-pods-outstanding] :or {command "exit 0"}} synthetic-pods-config]
+        (log/info "In" name "compute cluster, there are" num-synthetic-pods
+                  "outstanding synthetic pod(s), and a max of" max-pods-outstanding "are allowed")
+        (if (>= num-synthetic-pods max-pods-outstanding)
+          (log/info "In" name "compute cluster, cannot launch more synthetic pods")
+          (let [using-pools? (config/default-pool)
+                synthetic-task-pool-name (if using-pools? pool-name nil)
+                new-task-requests (remove (fn [{{:keys [job/uuid]} :job}]
+                                            (some #(= (str uuid) (controller/synthetic-pod->job-uuid %))
+                                                  outstanding-synthetic-pods))
+                                          task-requests)
+                task-metadata-seq (->>
+                                    new-task-requests
+                                    (map (fn [{{:keys [job/uuid] :as job} :job}]
+                                           {:task-id (str "synthetic-" pool-name "-" (d/squuid))
+                                            :command {:user user :value command}
```

Comment:
As mentioned earlier it would be great if the command output the id of the job for which this synthetic pod is created. We can include an environment variable for that job id and then configure the command in the setting to be an `echo` message of the environment variable.

Reply:
We should be able to map these back to their jobs with the pod label